### PR TITLE
Added test specs for e4s 23.05 and 23.08. 

### PR DESCRIPTION
**Pull request by wspear**
Fixed input parsing for testgen script (needed a 3rd argument, only checked for the first 2).

### CODE REVIEW

**Review by shahzebsiddiqui — CHANGES_REQUESTED** [review]
can you please add the template file and package list so we can see what packages were added.

File: buildspecs/e4s/E4S-Testsuite/perlmutter/23.05/py-warpx.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#This buildtest was generated from spack-test-23.05-template.yaml
+buildspecs:
+  py-warpx_spack-test_23.05:
```
Comment (shahzebsiddiqui) [inline buildspecs/e4s/E4S-Testsuite/perlmutter/23.05/py-warpx.yml:3]: please remove this test we dont have this installed in the `gcc` spack environment for e4s/23.05 i just checked

```console
siddiq90@login27> spack -e gcc find py-warpx
==> No package matches the query: py-warpx
```

**Comment by shahzebsiddiqui** [comment]
@wspear hold up i just realized we have spack tests for 23.08 in this PR https://github.com/buildtesters/buildtest-nersc/pull/240/ take a look at https://github.com/buildtesters/buildtest-nersc/tree/devel/buildspecs/e4s/spack_test/perlmutter. I see we have 23.05 tests too. I guess if you see any packages we have missed, we can add them.

**Comment by wspear** [comment]
Yeah, I thought that was already all done. I thought the spack_test path was for the spack internal tests so I didn't look for the e4s testsuite tests there before making the new PR.

**Comment by shahzebsiddiqui** [comment]
> Yeah, I thought that was already all done. I thought the spack_test path was for the spack internal tests so I didn't look for the e4s testsuite tests there before making the new PR.

I think lets close this PR and you can work on E4S Testsuite tests for these stacks see issue https://github.com/buildtesters/buildtest-nersc/issues/203. So only add test s for ones that have tests from E4S Testsuite. I dont know if there is an overlap with tests from spack test and E4S Testsuite for package X. If that is the case just simply skip from E4S Testsuite unless we see failure from spack test. 

Right now there are 50 failures see https://my.cdash.org/viewTest.php?onlyfailed&buildid=2452793 so its not looking good and we may want to resort to E4S Testsuite for some of these

File: buildspecs/e4s/E4S-Testsuite/perlmutter/23.05/aml.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#This buildtest was generated from spack-test-23.05-template.yaml
+buildspecs:
+  aml_e4s-testsuite_23.05:
+    type: spack
```
Comment (shahzebsiddiqui) [inline buildspecs/e4s/E4S-Testsuite/perlmutter/23.05/aml.yml:4]: change to script

**Pull request merged**